src/main/java/spiffe/api/svid/SpiffeEndpointChannelBuilder.java

```diff
@@ -0,0 +1,130 @@
+        if (isBlank(spiffeEndpointAddress)) {
+            spiffeEndpointAddress = getAddressFromEnv();
+        }
+
+        URI parsedAddress = parseAddress(spiffeEndpointAddress);
+        if (isTcp(parsedAddress)) {
+            return createTcpChannel(parsedAddress);
+        } else {
+            validateUDSAddress(parsedAddress);
+            return createNativeSocketChannel(parsedAddress);
+        }
+    }
+
+    /**
+     * Try to resolve the Address from the Environment.
+     * @throws IllegalStateException if the Address is not found
+     * @return the Address
+     */
+    private static String getAddressFromEnv() {
+        String address = System.getenv(ENV_ADDRESS_VAR);
+        if (isBlank(address)) {
+            throw new IllegalStateException(ENV_ADDRESS_VAR + " env var is not defined");
+        }
+        return address;
+    }
+
+    /**
+     * Create a TCP channel based on the given URI
+     *
+     * @param parsedAddress
+     * @return
+     */
+    private static ManagedChannel createTcpChannel(URI parsedAddress) {
+        return NettyChannelBuilder.forAddress(parsedAddress.getHost(), parsedAddress.getPort())
+                .negotiationType(NegotiationType.PLAINTEXT)
+                .build();
+    }
+
+    /**
+     * Create a Native Socket Channel pointing to the spiffeEndpointAddress
+     * @param spiffeEndpointAddress
+     * @return
+     */
+    private static ManagedChannel createNativeSocketChannel(URI spiffeEndpointAddress) {
+        NettyChannelBuilder channelBuilder = NettyChannelBuilder.
+                forAddress(new DomainSocketAddress(spiffeEndpointAddress.getPath()));
+        configureNativeSocketChannel(channelBuilder);
+        return channelBuilder
+                .usePlaintext()
+                .build();
+    }
+
+    /**
+     * Based on the detected OS, configures the Socket Channel EventLookGroup and Channel Type
+     *
+     * @param channelBuilder
+     */
+    private static void configureNativeSocketChannel(NettyChannelBuilder channelBuilder) {
+        if (SystemUtils.IS_OS_LINUX) {
+            channelBuilder.eventLoopGroup(new EpollEventLoopGroup())
+                          .channelType(EpollDomainSocketChannel.class);
+        } else if (SystemUtils.IS_OS_MAC) {
+            channelBuilder.eventLoopGroup(new KQueueEventLoopGroup())
+                          .channelType(KQueueServerDomainSocketChannel.class);
+        } else {
+            channelBuilder.eventLoopGroup(new NioEventLoopGroup());
+        }
+    }
+
+    private static URI parseAddress(String spiffeEndpointAddress) {
+        URI parsedAddress;
+        try {
+            parsedAddress = new URI(spiffeEndpointAddress);
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("The Spiffe Endpoint Address is not valid");
+        }
+        return parsedAddress;
+    }
+
+    private static void validateUDSAddress(URI address) {
+        if (!isBlank(address.getHost())) {
```

Comment:
maybe we can add guava preconditions, to avoid this lines and reduce it to something like 
Preconditions.checkState(!isBlank(address.getHost()), String.format("Unexpected Authority component in Unix uri: %s", address.getHost() ) );

Reply:
Applied your suggestion. Thanks!